tests/release_tests/test_launch/test_kubernetes.py

```diff
@@ -0,0 +1,187 @@
+import os
+import signal
+import sys
+from typing import Optional, Tuple
+
+import pytest
+from kubernetes import client, config, watch
+from utils import get_wandb_api_key, run_cmd, run_cmd_async
+from wandb.sdk.launch.launch_add import launch_add
+
+NAMESPACE = "wandb-release-testing"
+
+CONFIG = {
+    "resource_args": {
+        "kubernetes": {
+            "namespace": NAMESPACE,
+            "spec": {
+                "template": {
+                    "spec": {
+                        "containers": [
+                            {
+                                "env": [
+                                    {
+                                        "name": "WANDB_API_KEY",
+                                        "valueFrom": {
+                                            "secretKeyRef": {
+                                                "name": "wandb-api-key",
+                                                "key": "password",
+                                            }
+                                        },
+                                    },
+                                    {
+                                        "name": "AWS_ACCESS_KEY_ID",
+                                        "valueFrom": {
+                                            "secretKeyRef": {
+                                                "name": "aws-access-key-id",
+                                                "key": "password",
+                                            }
+                                        },
+                                    },
+                                    {
+                                        "name": "AWS_SECRET_ACCESS_KEY",
+                                        "valueFrom": {
+                                            "secretKeyRef": {
+                                                "name": "aws-secret-access-key",
+                                                "key": "password",
+                                            }
+                                        },
+                                    },
+                                    {
+                                        "name": "AWS_SESSION_TOKEN",
+                                        "valueFrom": {
+                                            "secretKeyRef": {
+                                                "name": "aws-session-token",
+                                                "key": "password",
+                                            }
+                                        },
+                                    },
+                                ]
+                            }
+                        ]
+                    }
+                }
+            },
+        }
+    },
+}
+
+
+def test_kubernetes_agent_on_local_process():
+    entity = "launch-release-testing"
+    project = "release-testing"
```

Comment:
is this project pre-made, or does it get created every time? if it already exists can we make it a constant? Also goes for the queue etc.

Reply:
The project, queue, and job are all pre-made. I'll make them constants.

Once we have programmatic queue creation, I'll consider making them passable as CLI args and have everything made on the fly, since currently there is a lot of setup involved if someone wants to run this on a custom base url.